### PR TITLE
Tweak lifecycle in EdgeCanvas to avoid setState() in render()

### PR DESCRIPTION
**Pull request by kwasi**
* Moved most of the state state into file level constants.
* Moved CSS into Typestyle to share measurements used both in stylesheets and in constructing the edge canvas.
* No visual changes.
* Updated snapshots to reflect new typestyle based class names

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/173)

### CODE REVIEW

**Comment by avdaredevil** [comment]
/lgtm
/approve

Thanks for getting this done @kwasi!

File: frontend/src/components/EdgeCanvas.tsx [hunk 1]
```diff
@@ -1,103 +1,94 @@
...
+const css = stylesheet({
+  edgeCanvas: {
+    border: 0,
+    display: 'block',
+    height: px(440),
+    marginLeft: px(cardBodyWidth),
+    marginTop: px(74),
+    overflow: 'visible',
+    position: 'absolute',
+    width: `${edgeWidth}px`,
```
Comment (avdaredevil) [inline frontend/src/components/EdgeCanvas.tsx]:
```suggestion
    width: px(edgeWidth),
```

Comment (kwasi) [reply frontend/src/components/EdgeCanvas.tsx]: Done.

**Comment by kwasi** [comment]
/test kubeflow-metadata-presubmit

Log from http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-metadata-presubmit-unittests-173-6d9cfa5-3858-ffef?tab=workflow&nodeId=kubeflow-metadata-presubmit-unittests-173-6d9cfa5-3858-ffef-1481247084&sidePanel=logs%3Akubeflow-metadata-presubmit-unittests-173-6d9cfa5-3858-ffef-1481247084%3Amain

ERROR: (gcloud.builds.submit) [Errno 104] Connection reset by peer
This may be due to network connectivity issues. Please check your network settings, and the status of the service you are trying to reach.
  (void)write(STDERR_FILENO, end_msg, strlen(end_msg));

**Pull request merged**